test/boost/aws_error_injection_test.cc

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "seastar/util/closeable.hh"
+#include "seastar/util/defer.hh"
+#include "test/lib/log.hh"
+#include "test/lib/random_utils.hh"
+#include "test/lib/scylla_test_case.hh"
+#include "test/lib/tmpdir.hh"
+#include "utils/exceptions.hh"
+#include "utils/s3/client.hh"
+#include <seastar/core/fstream.hh>
+#include <seastar/core/units.hh>
+#include <seastar/http/httpd.hh>
+
+using namespace seastar;
+using namespace std::chrono_literals;
+struct server {
+    enum class failure_policy : uint8_t {
+        SUCCESS = 0,
+        RETRYABLE_FAILURE = 1,
+        NONRETRYABLE_FAILURE = 2,
+    };
+    class dummy_aws_request_handler : public httpd::handler_base {
+    public:
+        explicit dummy_aws_request_handler(failure_policy _test_failure_policy) : test_failure_policy(_test_failure_policy) {}
+        future<std::unique_ptr<http::reply>> handle(const sstring& path, std::unique_ptr<http::request> req, std::unique_ptr<http::reply> rep) override {
+            auto method = req->_method;
+            auto url_params = req->query_parameters;
+            testlog.debug("{}\t{}", req->_method, req->get_url());
+            rep->write_body("txt", build_response(*req));
+            if (method == "DELETE")
```

Comment:
scylla and seastar coding style ~~suggests~~ insists to use {} even for one-line bodies

Reply:
Done